### PR TITLE
Account

### PR DESCRIPTION
**Pull request by eostendarp**
Account tests for non-trivial methods, and implementation

### CODE REVIEW

File: src/test/java/edu/ithaca/dragon/bank/AccountTest.java [hunk 1]
```diff
@@ -1,63 +1,513 @@
...
+        account = new CheckingAccount(115.42);
+        account.withdraw(115.42);
+        assertEquals(0, account.getBalance(), delta);
+        assertEquals("115.42 | w 115.42 | ", account.getTransactionHistory());
+
+        // withdraw 0 -boundary case
+        account = new CheckingAccount(200);
+        account.withdraw(0);
+        assertEquals(200, account.getBalance(), delta);
+        assertEquals("200.0 | w 0.0 | ", account.getTransactionHistory());
```
Comment (ConnorCurry) [inline src/test/java/edu/ithaca/dragon/bank/AccountTest.java]: Is there a reason we want to allow 0 amount withdrawal and deposits? I know we had a brief discussion about this as a whole class a little while back but we never really came to a consensus I don't think.

File: src/test/java/edu/ithaca/dragon/bank/AccountTest.java [hunk 2]
```diff
@@ -1,63 +1,513 @@
...
+        account = new CheckingAccount(200);
+        account.deposit(88.32);
+        assertEquals(288.32, account.getBalance(), delta);
+        assertEquals("200.0 | d 88.32 | ", account.getTransactionHistory());
+
+        // 0 amount -boundary case
+        account = new CheckingAccount(200);
+        account.deposit(0);
+        assertEquals(200, account.getBalance(), delta);
+        assertEquals("200.0 | d 0.0 | ", account.getTransactionHistory());
```
Comment (ConnorCurry) [inline src/test/java/edu/ithaca/dragon/bank/AccountTest.java]: Same as before with withdraw, should we allow 0 deposits?

File: src/test/java/edu/ithaca/dragon/bank/AccountTest.java [hunk 3]
```diff
@@ -1,63 +1,513 @@
...
+        assertEquals("200.0 | d 200.0 | ", to.getTransactionHistory());
+
+        // transfer 0 -boundary case
+        from = new CheckingAccount(200);
+        to = new CheckingAccount(200);
+        from.transfer(to, 0);
+        assertEquals(200, from.getBalance(), delta);
+        assertEquals("200.0 | w 0.0 | ", from.getTransactionHistory());
+        assertEquals(200, to.getBalance(), delta);
+        assertEquals("200.0 | d 0.0 | ", to.getTransactionHistory());
```
Comment (ConnorCurry) [inline src/test/java/edu/ithaca/dragon/bank/AccountTest.java]: Here as well

**Review by ConnorCurry — COMMENTED** [review]
This looks great. Extremely organized! I only had a couple questions about allowing 0 value deposits, withdrawals and transfers.

File: src/test/java/edu/ithaca/dragon/bank/AccountTest.java (hunk 1 above)
Comment (eostendarp) [reply src/test/java/edu/ithaca/dragon/bank/AccountTest.java]: My reasoning for allowing 0s was that a 0 amount doesn't cause issues in the same way that a negative amount might, but then there really isn't a reason for a user to want to use a 0 amount. I don't think there's any major advantages or disadvantages to allowing 0 amount withdraw, deposits, and transfers, but I can totally make the change for the sake of clarity.
